src/main/java/jp/ac/utokyo/rcast/karkinos/filter/SoftClipExtention.java

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright Hiroki Ueda
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package jp.ac.utokyo.rcast.karkinos.filter;
+
+import jp.ac.utokyo.rcast.karkinos.utils.TwoBitGenomeReader;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.SAMRecord;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SoftClipExtention {
+
+    //add 2020/12/17 for FFPE anneling near repeat, extends soft clipping
+    public static void extendSoftclip(SAMRecord sam, TwoBitGenomeReader tgr) {
+        try {
+            if(sam.getCigar().getCigarElements().size()>=2) {
+                leftExtention(sam, tgr);
+                rightExtention(sam, tgr);
+            }
+
+        }catch(Exception ex){
+            //could not extend soft clips
+            ex.printStackTrace();
+        }
+    }
+
+    private static void leftExtention(SAMRecord sam, TwoBitGenomeReader tgr) {
+        List<CigarElement> list = sam.getCigar().getCigarElements();
+        CigarElement first = list.get(0);
+        CigarElement second = list.get(1);
```

Comment:
This is meant for softclipped read only

Reply:
?? is checked by if(sam.getCigar().getCigarElements().size()>=2) {

Reply:
I'm sorry that I missed that. But rather than depending on an if statement outside, the method itself should be responsible for checking preconditions it requires for its arguments.